woocommerce/woocommerce-ios
# Refactor Rename Attributes VC to simplify and remove table view

Related: https://github.com/woocommerce/woocommerce-ios/pull/3768#pullrequestreview-600913106

## Description

This is an attempt to simplify `RenameAttributesViewController` by removing the table view and replacing it with a simplified layout that includes a text field and label. I created the layout in `TextFieldWithFooterView` so it could be reused in other VCs that are similarly simple (e.g. `ProductSlugViewController`).

This simplifies the view controller, but `TextFieldWithFooterView` does a fair amount of work to recreate a design that is based on table views, including some duplication of the logic in `TextFieldTableViewCell`. This was a useful exercise for me, but I'm curious if other think this refactor is worth it?

## Changes

* Adds `TextFieldWithFooterView`, which recreates the layout you'd get with a table view that has a single `TextFieldTableViewCell` and a footer. It puts the text field in a container view to provide an edge-to-edge background and border around the text field.
* Removes the table view from `RenameAttributesViewController` and replaces it with `TextFieldWithFooterView`.

Before|After
-|-
![Before](https://user-images.githubusercontent.com/8658164/110990405-304ca200-836b-11eb-8a9d-f17047310b9d.png)|![After](https://user-images.githubusercontent.com/8658164/110990410-317dcf00-836b-11eb-9f00-5327d0b906d9.png)


## Testing

1. Go to the Products tab.
2. Select a variable product.
3. Select Variations to open the variations list.
4. Tap the ... menu and select Edit Attributes.
5. Select a local attribute to open the Attribute Options screen.
6. Tap the ... menu and select Rename.
7. Confirm the Rename Attribute screen appears with the expected layout and behavior (can rename the selected attribute).

## Submitter Checklist

Update release notes:
